### ISSUE TITLE
Examples for training pipelines?

### ISSUE THREAD

**Issue by Nickersoft** [JohnSnowLabs/spark-nlp#2978]
**Name of the Spark NLP feature whose docs need improvement:**
Pipelines / LightPipeline 

**What you think the docs should say:**
Hey all, I'm relatively new to Spark NLP and currently looking into using LightPipeline to speed up my workflow (I'm using SparkNLP for small samples of text in a real-time production system, so faster inference is ideal). For most languages, the pretrained Explain DL pipelines seem to work, but for some unsupported languages (like Chinese or Japanese), it'd be helpful if the docs shed more light on training your own pipeline and saving it to disk via the `fit()` command. I was hoping the `spark-nlp-models` repo had some example training code, but it doesn't seem to :/

I think what is particularly confusing about training pipelines is, unlike training a NER model or something, where you are using a vast amount of training data to effectively train a prediction model, many pipelines either use pretrained or statistical annotators for their stages, which don't need training. 

So if you had a pipeline consisting of a DocumentAnnotator, Tokenizer, pretrained embedder and NER, and finisher, what data would be used to train a pipeline like that? What is even being trained, and does the training data matter? For the existing Explain DL pipelines, what data was used to train those?

I'm curious if there's something I'm missing in the docs, or if these use cases should be added to the docs. There is quite a bit of info on training annotators, and info on using pretrained pipelines, but no info on training pipelines themselves. The [only article I could find](https://medium.com/spark-nlp/spark-nlp-101-lightpipeline-a544e93f20f1), which touches briefly on training pipelines, runs a LightPipeline on a sentence from the same dataset used to train the pipeline, so not super helpful.

**Comment by maziyarpanahi** [comment]
Hi,

I think what's causing an issue for you is the term `.fit()` when you are using Pipeline in Spark/Spark NLP. Whether you have something to train inside your pipeline (let's say NerDLApproach) or not (everything is rule-based or already training) you still need to use `.fit()` before `.transform`. What happens in this situation is that `.fit()` will be skipped if there is nothing to train in the pipeline and it goes directly into `.transform()`. 

This is actually not a Spark NLP concept, you can find out more about this in Spark ML documentation since we are a native extension on top of that (so we are not duplicating those docs https://spark.apache.org/docs/latest/ml-pipeline.html). Every pipeline and model in Spark NLP is the same as Spark ML in concept. So if you have something in the pipeline that requires to be trained, it will take the DataFrame from `.fit(df)` and trains it, then you can save or immediately use it for prediction via `.transform(df2)`. If you have nothing to train and everything is either rule-based or already trained, then you can use the same DataFrame for both like `.fit(df).transform(df)` which in this case fit will skip the training part since there is nothing to train and goes to transform.

The pre-trained pipelines you are mentioning are constructed exactly the same way. We just stacked up some annotators like tokenizer, normalizer, etc. with pretrained models like NER or classification and then saved them as PipelineModel to be used by users via either `.transform` or `.annotate`. 

The `spark-nlp-models` is deprecated as the README mentions it. All the models/pipelines have moved to Models Hub .johnsnowlabs.com/models. 

This repo has all the examples and tutorials: https://github.com/JohnSnowLabs/spark-nlp-workshop

For instance, this notebook shows how to make a custom pipeline via pretrained models in Spark NLP: https://github.com/JohnSnowLabs/spark-nlp-workshop/blob/master/tutorials/Certification_Trainings/Public/3.SparkNLP_Pretrained_Models.ipynb
